src/compiler/scala/tools/nsc/typechecker/MethodSynthesis.scala

```diff
@@ -341,10 +343,13 @@ trait MethodSynthesis {
         case (p :: Nil) :: _  => p
         case _                => NoSymbol
       }
-      private def setterRhs = (
-        if (mods.isDeferred || derivedSym.isOverloaded) EmptyTree
+      // TODO: when is `derivedSym.isOverloaded`??? is it always an error?
+      // `noFieldHere` does not imply DEFERRED (ask isDeferred if you must know)
+      // we automatically implement trait setters (where !isDeferred) in the subclass
```

Comment:
This comment makes sense describing this patch, but I think it doesn't make much sense when reading the resulting code.